modules/Kconfig.nordic

```diff
@@ -11,6 +11,7 @@ menu "Nordic drivers"
 menuconfig NRF_802154_RADIO_DRIVER
 	bool "Enable nRF IEEE 802.15.4 radio driver"
 	depends on HAS_HW_NRF_RADIO_IEEE802154
+	select DYNAMIC_INTERRUPTS
```

Comment:
Can you elaborate on this change?

Reply:
The related PR to hal_nordic introduces an IRQ abstraction layer for the 802.15.4 driver. In its [Zephyr implementation](https://github.com/zephyrproject-rtos/hal_nordic/pull/49/files#diff-924ed59eeff09bff751d5d57c43d13e6), that abstraction layer is implemented with `irq_connect_dynamic`, as the usage of the module is not known at build time.

Reply:
OK , so you 're introducing a change to the 892.15.4 driver, so it now uses dynamic interrupts. This comes with an overhead on memory footprint, so I wonder if this could be configurable, and be selected for the use-cases you really need to install ISRs in run-time.

Reply:
Discussed this offline with @jciupis

It seems to me that the newer version of the 802.15.4 driver you are updating to _requires_ the DYNAMIC_INTERRUPTS feature as _mandatory_. As I said, this results in an inevitable overhead in memory footprint of nRF devices. However: if this is the way the driver shall work from now on, and since - from Zephyr's perspective - the driver is an external project, I do not object to the change, as long as the maintainer of the external project (in this case @hubertmis) confirms that this is the intended behavior.

To make this more clear, the `select DYNAMIC_INTERRUPTS` could, alternatively, be placed in the module's repository Kconfig (although it is also fine to be here).

If we keep it here, I would like the help text to explain briefly why Dynamic Interrupts are needed.

Reply:
@rlubos @carlescufi ^^

Reply:
@ioannisg , the external project is prepared as system-agnostic driver. We found it difficult to port it to Zephyr IRQ system and (as @rlubos requested some time ago), we added an abstraction layer for IRQ configuration to reduce risk of mistakes during the porting process, and to remove problems with direct NVIC configuration.

The intended behavior from the external project perspective is to work correctly within Zephyr RTOS. It looks that implemented IRQ configuration abstraction using dynamic IRQs meets this expectation. While the previous solution that were using NVIC macros directly by the driver was a source of problems in Zephyr.

However, taking into account your comment, we are looking for a way to make the IRQ configuration abstraction that we defined compliant with `IRQ_CONNECT` so that we won't have to use dynamic interrupts anymore. We need to asses what do we gain when disabling dynamic IRQs and what would be the cost of changing IRQ configuration abstraction. After we do that, we'll assess what is intended final behavior of the driver.

I would recommend to keep DYNAMIC_INTERRUPTS as a dependency of 802.15.4 driver for now, until we find a solution to use `IRQ_CONNECT` with platform abstraction. It is possible that we will keep it forever, if find that maintenance of IRQ_CONNECT-compliant platform abstraction is not worth cost of `DYNAMIC_INTERRUPTS`.

Reply:
That's fine. Just list the dependency then, in the help text with some reasoning.

Btw, the cost of dynamic interrupts, is a few API functions implementation (some text space) and a duplicate vector table in RAM (around 512 bytes in nRF5 series). And some extra ISR delay.

Reply:
@ioannisg , thanks for potential gain analysis! Now we need to asses only the cost.
@jciupis, please update the help text according to @ioannisg suggestion.